consensus/src/consensus_provider.rs

```diff
@@ -54,6 +55,8 @@ pub fn start_consensus(
 
     let (timeout_sender, timeout_receiver) = channel::new(1_024, &counters::PENDING_ROUND_TIMEOUTS);
     let (self_sender, self_receiver) = channel::new(1_024, &counters::PENDING_SELF_MESSAGES);
+    let shared_connections = Arc::new(RwLock::new(HashMap::new()));
+    network_sender.initialize(shared_connections.clone());
```

Comment:
Could the initialize step just be done in NetworkSender::new() and provide a getter that returns the shared_connections?  Then network_sender doesn't need to be mut?  Also, can that simplify ConsensusNetworkSender to have a required connections instead of Option?

Reply:
NetworkSender doesn't have a new function, there's a trait NewNetworkSender looks like this
```
pub trait NewNetworkSender {
    fn new(
        peer_mgr_reqs_tx: PeerManagerRequestSender,
        connection_reqs_tx: ConnectionRequestSender,
    ) -> Self;
}
```